### PR TITLE
Feat/feailed reasons connect

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Failed reason: Сall the endpoint to send the reason for the order failure

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-146)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/api/notificationActions.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  userId: number | undefined;
+}
+
+export const createNotification =
+  (payload: NotificationPayload) => async (): Promise<void> => {
+    try {
+      const response = await axiosInstance.post('/notifications', payload);
+
+      if (response.status === NOTIFICATION_SUCCESS_STATUS) {
+        toast.success(i18n.t(`${tBase}.notifications.success`));
```
Comment (Parle648) [inline src/api/notificationActions.ts]: why not just t()?

Comment (oyakovytskyi) [reply src/api/notificationActions.ts]: I just did the same thing as in the code with requests to endpoints. It seems that both options have almost the same functionality. If I change to import { t } from 'i18next', then replace it in other files as well?

Comment (Parle648) [reply src/api/notificationActions.ts]: You need to import only one method, not the entire module

Comment (oyakovytskyi) [reply src/api/notificationActions.ts]: replaced

**Review by AlexeyBezrebriy — APPROVED** [review]

**Review by Parle648 — APPROVED** [review]

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/pages/FailedReason/index.tsx [hunk 2]
```diff
@@ -23,6 +23,8 @@ import {
   tBase,
 } from '@/constants/constants';
 
+const orderId = 4;
```
Comment (lyudad) [inline src/pages/FailedReason/index.tsx]: I dont like that you hardrcode ID think about other option or write explanation

Comment (oyakovytskyi) [reply src/pages/FailedReason/index.tsx]: I just didn't see that the page from which the transition is made is ready, I fixed it

**Review by lyudad — APPROVED** [review]

**Pull request merged**